### DISCUSSION TITLE
bounds in a real-world map?

### DISCUSSION THREAD

**Discussion by Moonbase59** [javalent/obsidian-leaflet#95] [Ideas]
I wonder if bound could be put to sensible use in real-world maps, like for

- showing a rectangle overlay
- using specified bounds to fit that rectangle into the viewport, without distorting it

I use the `geocoder` library in Python, which brings a `geocode` cli app, and for instance a boundary, OSM-style, is returned for _lots_ of places, in these forms:

```
"bbox": {"northeast": [48.4245039, 10.3001662], "southwest": [48.4212526, 10.2950636]},
"boundingbox": ["48.4212526", "48.4245039", "10.2950636", "10.3001662"]
```

(JSON). Unfortunately, it uses the `[lat, lat, lon, lon]` format (southwest, northeast).

This could easily be queried and used in a note, for instance to set the viewport or to add a rectangle-shaped overlay.

Of course, the `mapoverlay` syntax would have to be modified to include the type of overlay, i.e.

```yaml
mapoverlay:
  - [circle, green, [48.4242278,10.3011045], 1.5 km, "Legoland Germany"]
  - [rectangle, blue, [48.4212526, 48.4245039, 10.2950636, 10.3001662], "Legoland Resort"]
  # — OR, giving northeast and southwest location: —
  - [rectangle, blue, [48.4245039, 10.3001662], [48.4212526, 10.2950636], "Legoland Resort"]
  # — OR, giving southwest and northeast location (more natural, I think): —
  - [rectangle, blue, [48.4212526, 10.2950636], [48.4245039, 10.3001662], "Legoland Resort"]
```
I don’t currently have a need for this, I only thought I’d mention it, so it’s not forgotten.

**Comment by valentine195** [comment]
Well, I wouldn't want to use `bounds`, since then you wouldn't be able to draw a box on the image map 😁 

I like `box`, I think, if I was going to add it. Then in-note syntax would be `mapbox`, so existing overlays wouldn't break.

**Reply by Moonbase59** [reply]
Maybe `bbox` and `mapbbox` or even `boundingbox` and `mapboundingbox`? This would more closely resemble the JSON values we get from 3rd-party software.

Btw, I personally prefer the "southwest/northeast" variant, although it’s more typing. Reason: If you come back weeks later, you’ll _still_ know what it means without much remembering, and the single locations could use the standard `location: [lat,lon]` format.

Also, if just using these two location pairs, it would be much easier for the `mapoverlay` syntax. I think.
